drivers/char/rust_example.rs

```diff
@@ -125,6 +128,59 @@ impl KernelModule for RustExample {
             cv.free_waiters();
         }
 
+        // Test threads.
+        {
+            let mut a = 1;
+            // XXX use a completion or Barrier
+            let flag = Arc::try_new(AtomicBool::new(false))?;
+            let other = flag.clone();
+
+            let t1 = Thread::new(
+                move || {
+                    other.store(true, Ordering::Release);
+                    let b = Box::try_new(42)?;
+                    for _ in 0..20 {
+                        a += 1;
+                        println!("Hello Rust Thread {}", a + b.as_ref());
+                    }
+
+                    Ok(())
+                },
+                cstr!("Rust thread"),
+            )?;
+
+            t1.wake_up();
+
+            // Wait to observe the thread run
```

Comment:
Nit: here and in all other comments. For consistency, please finish all sentences in with the appropriate punctuation, usually a period.